app/services/time_interval.rb

```diff
@@ -21,14 +21,27 @@ def week_end(week_start, length)
     end
   end
 
-  attr_reader :start, :end
+  attr_reader :starting, :ending
 
   def initialize(starting, ending)
-    @start = starting
-    @end = ending
+    @starting = starting
+    @ending = ending
   end
 
   def collect_steps(step)
-    (@start.to_i..@end.to_i).step(step).collect { |time| Time.at time }
+    (@starting.to_i..@ending.to_i).step(step).collect { |time| Time.at time }
+  end
+
+  def to_rfc3339
+    TimeIntervalRFC3339.new(starting, ending)
+  end
+
+  class TimeIntervalRFC3339
+    attr_reader :starting, :ending
+
+    def initialize(starting, ending)
+      @starting = starting.to_datetime.rfc3339(9)
```

Comment:
I'm not sure what the 9 number means :) its kind of formatter? You can crate static variable to keep magic 9 number. Something like

FORMATTER_NUMBER = 9 (or any other name)
private_constant :FORMATTER_NUMBER
and use here
@starting = starting.to_datetime.rfc3339(FORMATTER_NUMBER)

another than I like this